controller/src/main/java/org/jboss/as/controller/OperationContextImpl.java

```diff
@@ -1180,13 +1180,18 @@ void releaseStepLocks(AbstractOperationContext.Step step) {
                 try {
                     modelController.awaitContainerStability(timeout, TimeUnit.MILLISECONDS, true);
                 }  catch (InterruptedException e) {
+                    // Cancelled in some way
                     interrupted = true;
-                    MGMT_OP_LOGGER.interruptedWaitingStability();
+                    // Cancellation prevented us ensuring that MSC will be stable when this operation completes.
+                    // We can no longer have any sense of MSC state or how the model relates to the runtime and
+                    // we need to start from a fresh service container. Notify the controller of this.
+                    modelController.containerCannotStabilize();
```

Comment:
Originally I only did it for timeout. But what I saw is that in actual hang situations (where the timeout would trip), the HC's end up cancelling the op, resulting in this code path being the one that is followed. If we don't report here, then the report never happens. And it is true that if the op is allowed to complete before MSC stability is obtained, we no longer know the true state of the system.

I think the chances of 'false reporting' here are quite low. If MSC is unstable, rolling back probably won't stabilize it. (Instability is likely due to service start() or stop() not completing, and changes made by rollback, e.g. removing a service, don't repair that.) Perhaps in another PR I could tweak some of the timeout values to reduce the case where the HC cancels before the server times out here. And/or, I could add something where if another ops finds that MSC is stable again it triggers a new notification that "all is well".